### PR TITLE
New BitAt to get a bit from IBytes

### LINKED ISSUE #326: New BitAt to get a bit from byte array
### Expected Behavior

`BitAt` delivers the bit at a given position from a byte array.
The implementation is similar to `ItemAt`

### Actual Behavior

Not existing

### PR DESCRIPTION
**Pull request by aveXcaesar**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

close #326 

### What is the new behavior?

- new `BitAt()` implements `IScalar<byte>`

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

File: src/Yaapii.Atoms/Scalar/BitAt.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                        this.bytes
+                    );
+            }
+
+            return result;
+        }
+
+        private void FailWhenIndexIsNegative()
+        {
+            new FailWhen(() =>
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Scalar/BitAt.cs]: This is too much. Remove the fail and make a simple if please.

File: src/Yaapii.Atoms/Scalar/BitAt.cs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+                this.position < 0,
+                new ArgumentException($"The position must be non-negative but is {this.position}")
+            ).Go();
+        }
+
+        private bool CanGetBit(int byteIndex)
+        {
+            return this.bytes.AsBytes().Length > byteIndex;
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Scalar/BitAt.cs]: You are unnecessarily building the bytes two times inside this object. Please refactor.

File: src/Yaapii.Atoms/Scalar/BitAt.cs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            if (this.CanGetBit(byteIndex))
+            {
+                var relevantByte = this.bytes.AsBytes()[byteIndex];
+                result = (relevantByte & (1 << bitInByteIndex)) > 0;
+            }
+            else
+            {
+                result =
+                    this.fallback.Invoke(
+                        new ArgumentException($"Cannot get byte at position {byteIndex} because there are only {this.bytes.AsBytes().Length} bytes."),
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Scalar/BitAt.cs]: this code should be done by the secondary ctors. A default exception should be thrown by a fallback which is used if we have no custom exception specified.
It is currently irritating that we can have custom exception which will be thrown before this exception. This makes the class less readable.

**Comment by aveXcaesar** [comment]
@Meerownymous 
changes done

File: src/Yaapii.Atoms/Scalar/BitAt.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        { }
+
+        /// <summary>
+        /// The value of the first bit.
+        /// </summary>
+        /// <param name="bytes">Bytes from where the bit is taken</param>
+        /// <param name="fallback">Result in case of an error</param>
+        public BitAt(IBytes bytes, bool fallback) : this(
+            bytes,
+            new FuncOf<IBytes, bool>(itr =>
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Scalar/BitAt.cs]: please use c# internal lambda funcs, not IFunc interface

**Comment by aveXcaesar** [comment]
@Meerownymous
changes done

**Comment by aveXcaesar** [comment]
@Meerownymous
changes done

**Pull request merged**